### PR TITLE
Fix uninstall procedure.

### LINKED ISSUE #816: Plugin won't delete
### Describe the bug:

We're seeing multiple reports regarding being unable to delete the plugin. I've investigated the issues on some sites and could replicate it. The challenge is that this issue can't be replicated generally on all sites, it's specific to sites where there's an issue with the Pinterest connection (e.g. the connection is stuck and disconnecting via the plugin doesn't work)

I've found deleting the plugin's data in the database resolves the issue. i.e., delete the `pinterest_for_woocommerce` tables. 

### Open interactions

6708766-zd-a8c, 6737221-zd-a8c (AT)

### Other interactions 

6369013-zd-a8c - Removing plugin data from the DB resolved the issue
https://wordpress.org/support/topic/feed-and-plugin-wont-delete - They had to delete via FTP
6194365-zd-a8c - Similar issue but specific to the connection issue. (The plugin could be deleted in this specific instance)

### Similiar Github issues:

https://github.com/woocommerce/pinterest-for-woocommerce/issues/363
https://github.com/woocommerce/pinterest-for-woocommerce/issues/779

### Steps to reproduce:

1. Access this ticket 6737221-zd-a8c
2. Access the AT site
3. Deactivate and attempt to delete the Pinterest plugin (Notice it just loads....)

### Logs

Affected sites have this fatal error:

```
2023-08-27T10:59:57+00:00 CRITICAL Uncaught Error: Class 'Automattic\WooCommerce\Pinterest\FeedRegistration' not found in /wordpress/plugins/pinterest-for-woocommerce/1.3.8/uninstall.php:25
Stack trace:
#0 /wordpress/core/6.3/wp-admin/includes/plugin.php(1245): include_once()
#1 /wordpress/core/6.3/wp-admin/includes/plugin.php(960): uninstall_plugin('pinterest-for-w...')
#2 /wordpress/core/6.3/wp-admin/includes/ajax-actions.php(4732): delete_plugins(Array)
#3 /wordpress/core/6.3/wp-includes/class-wp-hook.php(310): wp_ajax_delete_plugin('')
#4 /wordpress/core/6.3/wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters('', Array)
#5 /wordpress/core/6.3/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#6 /wordpress/core/6.3/wp-admin/admin-ajax.php(188): do_action('wp_ajax_delete-...')
#7 {main}
  thrown in /wordpress/plugins/pinterest-for-woocommerce/1.3.8/uninstall.php on line 25
```

I also found this fatal error log on the linked AT site:

```
2023-08-28T18:01:49+00:00 CRITICAL Uncaught TypeError: Argument 1 passed to Automattic\WooCommerce\Pinterest\Admin\Product\Attributes\VariationsAttributes::get_form() must be an instance of WC_Product_Variation, bool given, called in /wordpress/plugins/pinterest-for-woocommerce/1.3.8/src/Admin/Product/Attributes/VariationsAttributes.php on line 116 and defined in /wordpress/plugins/pinterest-for-woocommerce/1.3.8/src/Admin/Product/Attributes/VariationsAttributes.php:143
Stack trace:
#0 /wordpress/plugins/pinterest-for-woocommerce/1.3.8/src/Admin/Product/Attributes/VariationsAttributes.php(116): Automattic\WooCommerce\Pinterest\Admin\Product\Attributes\VariationsAttributes->get_form(false, 15)
#1 /wordpress/plugins/pinterest-for-woocommerce/1.3.8/src/Admin/Product/Attributes/VariationsAttributes.php(69): Automattic\WooCommerce\Pinterest\Admin\Product\Attributes\VariationsAttributes->handle_save_variation(13699, 15)
```

### PR DESCRIPTION
**Pull request by budzanowski**
### Changes proposed in this Pull Request:

Closes #816 .

The autoloader call was missing hence code used in the uninstall procedure was not found.
We need to use that code because part of a successful uninstall is a cleanup on the Pinterest side.

### Detailed test instructions:

1. Use #816 for testing instructions

### Changelog entry

Fix -  Uninstall procedure.

### CODE REVIEW

**Comment by budzanowski** [comment]
Testing zip:
[pinterest-for-woocommerce.zip](https://github.com/woocommerce/pinterest-for-woocommerce/files/12485940/pinterest-for-woocommerce.zip)

File: uninstall.php [hunk 1]
```diff
@@ -14,15 +14,23 @@
...
 }
 
-/**
- * Remove the feed configuration.
- */
-$data        = get_option( 'pinterest_for_woocommerce_data', [] );
-$merchant_id = $data['merchant_id'] ?? '';
+try {
+	// Load classes.
+	require_once __DIR__ . '/pinterest-for-woocommerce.php';
```
Comment (JPry) [inline uninstall.php:19]: Do we need to load the main plugin file, or can we just include the autoloader?

Comment (budzanowski) [reply uninstall.php:19]: We need it.
There are constants that are required by the cleanup code and some other parts are needed as well.

I was looking for a cleaner way to do it but was not able to find anything without a bigger refactor.
The problem is that we need to do some API calls that clean up things on the Pinterest side.
This code will allow the uninstall procedure to go through to the end. The plugin will be deleted - even if things will not be cleaned 100%, but at least people will get rid of the plugin.

**Review by JPry — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**